keps/sig-release/3744-stay-on-supported-go-versions/README.md

```diff
@@ -0,0 +1,416 @@
+Even though many of these issues were not relevant to Kubernetes, many were,
+so it remained important to use supported go versions that received those fixes.
+
+An obvious solution would be to simply update Kubernetes release branches to new minor versions of go.
+However, Kubernetes also [avoids destabilizing changes in patch releases](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-release/cherry-picks.md#what-kind-of-prs-are-good-for-cherry-picks),
+and historically, this prevented updating release branches to new minor versions of go,
+due to changes that were considered prohibitively complex, risky, or breaking. Examples include:
+
+* go1.6: enabling http/2 by default
+* go1.13: go module build and code generation changes
+* go1.14: EINTR handling ([#92521](https://github.com/kubernetes/kubernetes/issues/92521))
+* go1.17: dropping x509 CN support, ParseIP changes
+* go1.18: disabling x509 SHA-1 certificate support by default
+* go1.19: dropping current-dir LookPath behavior
+
+Some of these changes could be easily mitigated in Kubernetes code,
+some could only be opted out of via a user-specified GODEBUG envvar,
+and others required invasive code changes or could not be avoided at all.
+
+Because of this inconsistency, Kubernetes release branches historically remained on a single go minor version,
+and risked being unable to pick up go security fixes for the last several months of each Kubernetes minor version.
+
+Feedback to the go team about this situation prompted a [discussion](https://github.com/golang/go/discussions/55090),
+[proposal](https://github.com/golang/go/issues/56986), [talk at GopherCon](https://www.youtube.com/watch?v=v24wrd3RwGo),
+and a [design](https://go.dev/design/56986-godebug) for improving backward compatibility in go in ways that would allow
+projects like Kubernetes to use supported go versions while retaining previous go version runtime behavior for a period of time.
+
+With that proposal [likely to be accepted](https://github.com/golang/go/issues/56986#issuecomment-1371307658),
+and a [request to clarify Kubernetes' approach to updating go versions on release branches](https://github.com/kubernetes/kubernetes/issues/112408),
+it seemed like a good time to capture requirements and a process for updating Kubernetes release branches.
+
+### Goals
+
+* improve security of Kubernetes patch releases by building and releasing using a supported minor version of go
+* preserve stability of Kubernetes patch releases by avoiding regressions or behavior changes due to updating go minor versions
+* avoid end-user action-required items in patch releases due to updating go minor versions
+
+### Non-Goals
+
+* change the cadence of Kubernetes patch releases
+* change the duration of support for a given Kubernetes minor version
+
+## Proposal
+
+Track changes made to the Kubernetes master branch that were required to adopt a new go minor version (go 1.N).
```

Comment:
Is this a viable tweak?
```suggestion
Track changes made to the Kubernetes default branch that were required to adopt a new go minor version (go 1.N).
```
It'd let us leave the wording intact even if we rename the default branch.

Reply:
sure, pushed a rephrase